code/game/objects/structures/railing.dm

```diff
@@ -0,0 +1,188 @@
+
+/obj/structure/railing/attack_hand(mob/user) //Here's my own little addition. Click a railing while pulling something to dump it over. Nukeops, get in that locker, it's time to drop into the captain's office.
+	. = ..()
+	if(user.pulling)
+		to_chat(user, "<span class='warning'>You start to dump [user.pulling] over [src]!</span>")
+		if(do_after(user, 50, target = src))
+			if(!user.pulling || QDELETED(user.pulling))
+				return
+			visible_message("<span class='warning'>[user] dumps [user.pulling] over [src]!</span>")
+			user.pulling.forceMove(get_step(src, src.dir))
+
+/obj/structure/railing/built //Player constructed
+	anchored = FALSE
+
+/obj/structure/railing/attackby(obj/item/I,mob/user)
+	if(default_unfasten_wrench(user, I))
+		update_icon()
+		return FALSE
+	. = ..()
+
+/obj/structure/railing/AltClick(mob/user)
+	. = ..()
+	revrotate() //Rotate clockwise
+
+/obj/structure/railing/Initialize(constructed=0)
+	. = ..()
+	if(anchored)
+		START_PROCESSING(SSobj,src)
+		update_icon(0)
+
+/obj/structure/railing/process()
+	. = ..()
+	update_icon(0)
+
+/obj/structure/railing/Destroy()
+	anchored = null
+	broken = 1
+	for(var/obj/structure/railing/R in oview(src, 1))
+		R.update_icon()
+	. = ..()
+
+/obj/structure/railing/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(!mover)
+		return 1
+	if(get_dir(loc, target) == dir)
+		return !density
+	else
+		return 1
+
+/obj/structure/railing/proc/NeighborsCheck(var/UpdateNeighbors = 1)
+	check = 0
+	//if (!anchored) return
+	var/Rturn = turn(src.dir, -90)
+	var/Lturn = turn(src.dir, 90)
+
+	for(var/obj/structure/railing/R in src.loc)// ������ ������, ��� ��������� ��� ������
+		if ((R.dir == Lturn) && R.anchored)//�������� ����� �������
+			//src.LeftSide[1] = 1
+			check |= 32
+			if (UpdateNeighbors)
+				R.update_icon(0)
+		if ((R.dir == Rturn) && R.anchored)//�������� ������ �������
+			//src.RightSide[1] = 1
+			check |= 2
+			if (UpdateNeighbors)
+				R.update_icon(0)
+
+	for (var/obj/structure/railing/R in get_step(src, Lturn))//������ ����� ������ �� ����������� �������
+		if ((R.dir == src.dir) && R.anchored)
+			//src.LeftSide[2] = 1
+			check |= 16
+			if (UpdateNeighbors)
+				R.update_icon(0)
+	for (var/obj/structure/railing/R in get_step(src, Rturn))//������ ������ ������ �� ����������� �������
+		if ((R.dir == src.dir) && R.anchored)
+			//src.RightSide[2] = 1
+			check |= 1
+			if (UpdateNeighbors)
+				R.update_icon(0)
+
+	for (var/obj/structure/railing/R in get_step(src, (Lturn + src.dir)))//������ ��������-����� ��������� ������������ ����������� �������.
+		if ((R.dir == Rturn) && R.anchored)
+			check |= 64
+			if (UpdateNeighbors)
+				R.update_icon(0)
+	for (var/obj/structure/railing/R in get_step(src, (Rturn + src.dir)))//������ ��������-������ ��������� ������������ ����������� �������.
+		if ((R.dir == Lturn) && R.anchored)
+			check |= 4
+			if (UpdateNeighbors)
+				R.update_icon(0)
+
+/obj/structure/railing/update_icon(var/UpdateNeighgors = 1)
```

Comment:
holy shit all of this is called per tick because of processing what the fuck is this code